routemaster/middleware/client_authentication.rb

```diff
@@ -14,10 +15,8 @@ def call(env)
 
       private
 
-      def _authenticate(username, _password)
-        @_users ||= Set.new(
-          ENV.fetch('ROUTEMASTER_CLIENTS', '').split(','))
-        !! @_users.include?(username)
+      def _authenticate(uuid, _username)
+        !! Models::ClientToken.get_all.has_key?(uuid)
```

Comment:
Materialising all keys and then checking for presence is going to be more expensive than checking for the presence of a single key on the server. I'd be inclined to make this a `exist?` method on `Models::ClientToken` so the implementation is hidden and then either cache the key set and query it locally, or just query a single key from Redis. This will be called a lot so we should make sure it's as performant as possible.